### PR TITLE
Feature/#44 Errror Boundaries

### PR DESCRIPTION
**Pull request by flxtagi**
Draft: Error boundaries. 
Works for some but not all.

### CODE REVIEW

**Comment by rudigier** [comment]
Sagst bescheid, wenn ich den reviewen kann. Steht ja noch auf "This pull request is still a work in progress"

**Comment by flxtagi** [comment]
> Sagst bescheid, wenn ich den reviewen kann. Steht ja noch auf "This pull request is still a work in progress"

ist ready for review, bevor wir mergen noch den main rein damit keine merge-conflicts auftreten.

**Comment by rudigier** [comment]
Would stay more on the default and do it like this: 

Component
- ErrorMessage (status, headline, message)

Page
404 (User ErrorMessage Component)
500 (User ErrorMessage Component)

If Fetch does not bring the data we need show ErrorMessage Component instead of fake error pages (fake, because we do net set HTTP status to 404 or 500)

**Comment by rudigier** [comment]
probably we could use:
import ErrorPage from 'next/error'

(...)

if (someError) {
    return <ErrorPage statusCode={500} />;
}

**Comment by flxtagi** [comment]
> it is quiet standard
https://nextjs.org/docs/advanced-features/error-handling

the only thing custumized is the layoutcomponent wrapper. which is just nice mumble look and feel but standard serverside error 500 or client side error. (and your suggested user-not found error)
https://nextjs.org/docs/advanced-features/custom-error-page#customizing-the-500-page

File: src/pages/global-error.tsx
Comment (flxtagi) [inline src/pages/global-error.tsx:1]: this is the global error: kinda last resort. if no network and no context could be read and / or also the router is failing. so this is a basic html rendering with a error output. -> better than browser output page.

File: src/pages/global-error.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+			<head>Mumble Global Error Page</head>
+			<body>
+				<h2>Something went really wrong!</h2>
+				<p>Try to screenshot this page and send it to your friendly printer-fixer admin</p>
+				<br />
+				<p>Error Message: {error?.message}</p>
+				<p>Error Stack: {error?.stack}</p>
+				<br /> ¯\_(ツ)_/¯
+				<br />
+				<a href="/">Go back home</a>
```
Comment (flxtagi) [inline src/pages/global-error.tsx:19]: login page link

File: src/pages/user/[id].tsx [hunk 2]
```diff
@@ -31,14 +32,7 @@ const UserPage: FC<TUserPage> = ({ user, posts, likes }: InferGetServerSideProps
...
 	if (!user) {
-		// TODO: better content or 404 page with nice illustration of a lost user
-		return (
-			<MainLayout>
-				<div className="text-slate-900 text-center">
-					<Headline level={3}>User not found</Headline>
-				</div>
-			</MainLayout>
-		);
+		return <FourOhFourPage error={Error} reason={'missing User'} />;
```
Comment (flxtagi) [inline src/pages/user/[id].tsx:35]: passing Error reason to global error or another generic site

File: src/components/ErrorBoundary.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	}
+	static getDerivedStateFromError(error: unknown) {
+		return { hasError: true, error };
+	}
+
+	componentDidCatch(error: Error | unknown) {
+		// may we also introduce an error logging service here - if there is time for that
+		console.error({ error });
+		const message = error?.toString() || 'unknown';
+		return <FourOhFourPage error={Error} reason={message} />;
```
Comment (flxtagi) [inline src/components/ErrorBoundary.tsx:29]: remove return to fourOhFour.

Comment (flxtagi) [reply src/components/ErrorBoundary.tsx]: done

**Review by rudigier — APPROVED** [review]

**Review by rudigier — APPROVED** [review]

**Pull request merged**